src/modules.rs

```diff
@@ -0,0 +1,182 @@
+// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+use crate::ansi;
+use crate::deno_dir::DenoDir;
+use crate::libdeno::deno_mod;
+use crate::msg;
+use std::collections::HashMap;
+use std::collections::HashSet;
+use std::fmt;
+
+pub struct ModuleInfo {
+  name: String,
+  children: Vec<deno_mod>,
+}
+
+/// A collection of JS modules.
+pub struct Modules {
+  pub info: HashMap<deno_mod, ModuleInfo>,
+  pub by_name: HashMap<String, deno_mod>,
+}
+
+impl Modules {
+  pub fn new() -> Modules {
```

Comment:
you should consider deriving a `Default` implementation for `modules::Modules`

note: #[warn(clippy::new_without_default)] on by default
help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#new_without_default

Reply:
Fixed.